great_expectations/datasource/fluent/dynamic_pandas.py

```diff
@@ -158,6 +159,8 @@ def _replace_builtins(input_: str | type) -> str | type:
             ),
         )
     },
+    # sql
+    "con": {"con": _FieldSpec(Union[ConfigStr, str], ...)},  # type: ignore[arg-type]
```

Comment:
@NathanFarmer this is what I did for "con"

Reply:
Only if we want to be able to do the config substitution. 
So things like `connect_str`, credentials etc.